cmd/repo-brancher/main.go

```diff
@@ -235,3 +204,33 @@ func main() {
 		logrus.WithError(err).Fatal("Could not branch configurations.")
 	}
 }
+
+func executeGitCMDFactory(dir string) func(l *logrus.Entry, args ...string) error {
+	return func(l *logrus.Entry, args ...string) error {
+		l = l.WithField("commands", fmt.Sprintf("git %s", strings.Join(args, " ")))
+		var b []byte
+		var err error
+		l.Debug("Running command.")
+		sleepyTime := time.Second
+		for i := 0; i < 3; i++ {
+			c := exec.Command("git", args...)
+			c.Dir = dir
+			b, err = c.CombinedOutput()
+			if err != nil {
+				err = fmt.Errorf("running git %v returned error %w with output %q", args, err, string(b))
+				l.WithError(err).Debugf("Retrying #%d, if this is not the 3rd try then this will be retried", i+1)
+				time.Sleep(sleepyTime)
+				sleepyTime *= 2
+				continue
+			}
+			break
+		}
+		l = l.WithField("output", string(b))
+		if err != nil {
+			l.Error("Failed to execute command.")
+		} else {
+			l.Debug("Executed command.")
+		}
+		return err
```

Comment:
Can we just do this instead ?
```suggestion
		if err != nil {
			l.Error("Failed to execute command.")
			return err
		} 
		l.Debug("Executed command.")
		return nil
```

Reply:
good point, updated